lib/phoenix_live_view/controller.ex

```diff
@@ -111,6 +113,23 @@ defmodule Phoenix.LiveView.Controller do
     end)
   end
 
+  defp put_new_layout_from_router(conn) do
+    if router = conn.private[:phoenix_router] do
```

Comment:
We actually want to compute this at compile-time in the router. So when you call the `live` macro, you can compute this and then you generate a route that passes the inflected router as a private, like we already pass general options:

   private: %{phoenix_live_view: unquote(opts)},

And then you can unpack it here. :)